libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/Store.kt

```diff
@@ -68,20 +69,25 @@ open class Store {
     }
 
     fun dispatch(action: Action) {
-        if (isDispatching) {
-            logW(
-                "Cannot dispatch new actions during reducer processing. " +
-                    "Action dropped: ${action::class.java.simpleName}.",
-                "Store"
-            )
-            return
+        actions.add(action)
+        if (isProcessing) return
+
+        isProcessing = true
+        val processedActions = hashSetOf<Action>()
+        while (actions.isNotEmpty()) {
+            val head = actions.remove()
+            if (!processedActions.add(head)) {
+                logW(category = "Store") {
+                    "Possibly infinite loop detected. Current action: $action, " +
```

Comment:
Should this crash? It seems we should just make it against the rules to create actions that trigger the same action over again 🤔

Reply:
Otherwise it is just dropping one of the actions in a trail of actions

Reply:
It is dropping the whole trail. I don't know, maybe crashing is too extreme, besides we will see these logs in Firebase.